go/arrow/array/half_float.go

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package array
+
+import (
+	"fmt"
+	"github.com/apache/arrow/go/arrow"
+	"strings"
+)
+
+// A type which represents an immutable sequence of Float16 values.
+type HalfFloat struct {
+	array
+	values []arrow.Float16
+}
+
+func NewHalfFloatData(data *Data) *HalfFloat {
+	a := &HalfFloat{}
+	a.refCount = 1
+	a.setData(data)
+	return a
+}
+
+func (a *HalfFloat) Value(i int) float32 { return a.values[i].Float32() }
```

Comment:
why do we convert it to `float32`?
couldn't we be lazy and just return the `float16` unchanged? and let the user do what she needs?

Reply:
Can we provide two methods? one returns float16, the other returns float32